encoding/thrift/register.go

```diff
@@ -79,6 +79,10 @@ type Service struct {
 	Name          string
 	Methods       map[string]UnaryHandler
 	OnewayMethods map[string]OnewayHandler
+
+	// Snippet of Go code representing the function definition of the different
+	// handlers. This is useful for introspection.
+	Signatures map[string]string
```

Comment:
We use thrift.*Handler(s) as function/method wrappers. Adding a Signature() method would then require to build some structure during code-gen to hold the pair (function, signature) which is a bigger change to code-gen. But that's arguable, @abhinav what are your thoughts?